Compile issue in Entity class due to conversion error

#### Describe the bug

Project doesn't compile.

Compiler output
```
Hazel\Hazel\src\Hazel\Scene\Entity.h(43,49): error C2676: binary '!=': 'const entt::entity' does not define this operator or a conversion to a type acceptable to the predefined operator (compiling source file src\EditorLayer.cpp)
```

#### To Reproduce
Steps to reproduce the behavior:
1. Re-clone the entire project from GitHub
1. Try to compile :)
2. See error above

#### Screenshots
https://github.com/TheCherno/Hazel/blob/5658efcc6b91c02c8a6528bdd6f89ef5347fae4f/Hazel/src/Hazel/Scene/Entity.h#L43

Type conversion error :
![2020-07-23 14_19_15-Window](https://user-images.githubusercontent.com/24603909/88285537-91960700-ccef-11ea-8c9c-c9fb90c8802c.png)

#### Operating system:
- OS: win-64
- Version: Windows 10 home

#### Proposed fix
In PR #285

--- a/Hazel/src/Hazel/Scene/Entity.h
+++ b/Hazel/src/Hazel/Scene/Entity.h
@@ -40,7 +40,7 @@ namespace Hazel {
 			m_Scene->m_Registry.remove<T>(m_EntityHandle);
 		}
 
-		operator bool() const { return m_EntityHandle != 0; }
+		operator bool() const { return static_cast<int>(m_EntityHandle) != 0; }
 	private:
 		entt::entity m_EntityHandle{ 0 };
 		Scene* m_Scene = nullptr;

--- a/Hazel/src/Hazel/Scene/Entity.h
+++ b/Hazel/src/Hazel/Scene/Entity.h
@@ -40,9 +40,9 @@ namespace Hazel {
 			m_Scene->m_Registry.remove<T>(m_EntityHandle);
 		}
 
-		operator bool() const { return static_cast<int>(m_EntityHandle) != 0; }
+		operator bool() const { return m_EntityHandle != entt::null; }
 	private:
-		entt::entity m_EntityHandle{ 0 };
+		entt::entity m_EntityHandle{ entt::null };
 		Scene* m_Scene = nullptr;
 	};
 

